app/src/main/java/com/example/jwplayerdemo/jwsettings/JWViewModel.java

```diff
@@ -56,76 +74,120 @@
      * @see JWPlayerViewSettings#onOptionsItemSelected(android.view.MenuItem)
      */
     void setCurrentConfig() {
-        Log.i("HYUNJOO", "JWViewModel - setupConfig() " + file);
+        Log.i("HYUNJOO", "JWViewModel - setupConfig() ");
         selected.setValue(setupConfig());
     }
 
     /*
      * Setup JW Config
      * */
     private PlayerConfig setupConfig() {
-        Log.i("HYUNJOO", "JWViewModel - getConfig(): " + file);
-
-        return config = new PlayerConfig.Builder()
-//                .advertising()
-//                .allowCrossProtocolRedirects()
-//                .autostart()
-//                .captionsConfig()
-//                .controls()
-//                .displayDescription()
-//                .displayTitle()
-                .file(file)
-//                .image()
-//                .logoConfig()
-//                .mute()
-//                .nextUpDisplay()
-//                .nextUpOffset()
-//                .playbackRates()
-//                .playlist()
-//                .preload()
-//                .relatedConfig()
-//                .repeat()
-//                .sharingConfig()
-//                .skinConfig()
-//                .stretching()
-//                .useTextureView()
-                .build();
+        Log.i("HYUNJOO", "JWViewModel - getConfig() ");
+
+        List<PlaylistItem> playlistItemList = new ArrayList<PlaylistItem>(){{
+            add(playlistItem);
+        }};
+
+        config.setPlaylist(playlistItemList);
+
+        return config;
     }
 
     void setFile(String file) {
         JWLogger.log("JWViewModel - Add stream tag: " + file);
-
         if (file.startsWith("{")) {
             JWLogger.log("JWViewModel - Add JSON tag: " + file);
-            this.json = file;
-        } else {
-            this.file = file;
-        }
+            config.setPlaylist(PlaylistItem.listFromJson(file));
+        } else config.setFile(file);
     }
 
     void setMediaId(String mediaid) {
-        JWLogger.log("JWViewModel - Add Media Id: " + mediaid);
     }
 
     void setPlaylistId(String playlistid) {
-        JWLogger.log("JWViewModel - Add Playlist Id: " + playlistid);
     }
 
     void setTitle(String title) {
         JWLogger.log("JWViewModel - Add Title: " + title);
+        playlistItem.setTitle(title);
+        config.setDisplayTitle(checkedTitle);
+    }
+
+    void setDescription(String description) {
+        JWLogger.log("JWViewModel - Add Description: " + description);
+        playlistItem.setDescription(description);
+        config.setDisplayDescription(checkedDescription);
     }
 
     void setImage(String image) {
         JWLogger.log("JWViewModel - Add Image: " + image);
+        if (checkedImage) playlistItem.setImage(image);
```

Comment:
No need to change this but some interesting reading around it:
https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/
https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492

Reply:
Oh wow! I didn't know! I thought shorter was always better! I will gladly change this to indentation and avoid bugs and crashes!

Reply:
I have another question is this better than using switch statements?